src/main.c

```diff
@@ -1313,13 +1316,78 @@ void WINAPI EventCallback(PEVENT_RECORD ev)
     }
 }
 
+int GetDefaultOutFileName(const wchar_t* inFileName, wchar_t* outBuffer, size_t outBufferCapacity)
+{
+    const size_t outFileExtensionLength = ARRAYSIZE(DEFAULT_OUT_FILE_EXTENSION); // Character count of file extension including null terminator.
+    int Err = NO_ERROR;
+    unsigned int inFilePathLength = 0;
+    size_t nameLengthWithoutExtension = 0;
+    size_t maxLengthWithoutExtension = 0;
+    wchar_t inFilePath[MAX_PATH] = { 0 };
+    wchar_t* inFileExtensionStart = NULL;
+
+    maxLengthWithoutExtension = outBufferCapacity - outFileExtensionLength;
+
+    // Assuming long paths have not been enabled, we need to work with the full path of the input file
+    // to avoid going over the MAX_PATH limit.
+    inFilePathLength = GetFullPathName(inFileName, ARRAYSIZE(inFilePath), inFilePath, NULL);
+    if (inFilePathLength >= ARRAYSIZE(inFilePath)) {
+        Err = ERROR_INVALID_PARAMETER;
+        printf("Absolute path length for input file is %u. Please provide an input file with an absolute path less than %u characters long.\n",
```

Comment:
Did you actually run into an issue with MAX_PATH, or is this more like rigor?

The input file ext will practically always be .etl, which means only when it's within 3 chars of the max path length will we run into issues here. And in that case, wouldn't CreateFile return an error code that makes the problem clear? (i.e., instead of this complicated function, can't we just print a helpful error message and bail when CreateFile returns that error code?)

Reply:
I ended up going down this path because I got some unexpected results when I was trying to validate the max buffer size on my machine.

CreateFile's GetLastError() call returns 3 (seems to be ERROR_PATH_NOT_FOUND) when the absolute path is longer than MAX_PATH, even when filenames with longer paths can be created on the same disk via other means.

[Reading up on CreateFileW](https://learn.microsoft.com/en-us/windows/win32/api/fileapi/nf-fileapi-createfilew#parameters), it looks like we could prepend `\\?\` to the output file path in order to use absolute paths with more than 260 characters.

I'll look a bit further into that and try to simplify this. I agree that it's pretty complicated for the task at hand.

Reply:
Whew, this is still more code than I expected but it doesn't seem to me like making any of the obvious simplifying assumptions (e.g. only allowing input files with ".etl" extensions) really cleans up the code, so I'd say after changing the camelCase to PascalCase we are good to go here. Thanks again!

Reply:
Yeah, honestly it was a bit more code than I expected too. Thanks for going over everything and taking on the new feature. This tool has helped me a lot, and I'm glad to be able to contribute!